Buildings/Applications/DHC/CentralPlants/Cooling/Controls/ChillerStage.mo

```diff
@@ -0,0 +1,152 @@
+within Buildings.Applications.DHC.CentralPlants.Cooling.Controls;
+model ChillerStage "Stage controller for chillers"
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Modelica.SIunits.Time tWai "Waiting time";
+  parameter Modelica.SIunits.Power QEva_nominal
+    "Nominal cooling capaciaty (negative)";
+  parameter Modelica.SIunits.Power  criPoiLoa = 0.55*QEva_nominal
+    "Critical point of cooling load for switching one chiller on or off";
+  parameter Modelica.SIunits.Power  dQ = 0.25*QEva_nominal
+    "Deadband for critical point of cooling load";
+
+  Modelica.Blocks.Interfaces.RealInput QLoa(unit="W",max=0)
+    "Total cooling load, negative"
+    annotation (Placement(transformation(extent={{-140,-60},{-100,-20}})));
+  Modelica.Blocks.Interfaces.BooleanInput on "On signal of the chillers"
+    annotation (Placement(transformation(extent={{-140,20},{-100,60}})));
+  Modelica.Blocks.Interfaces.RealOutput y[2]
+    "On/off signal for the chillers - 0: off; 1: on"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+  Modelica.StateGraph.InitialStep off(nIn=1) "No cooling is demanded"
+    annotation (Placement(transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=-90,
+        origin={-50,70})));
+  Modelica.StateGraph.StepWithSignal oneOn(nOut=2, nIn=2)
+    "Status of one chiller on"
+    annotation (Placement(
+        transformation(
+        extent={{10,-10},{-10,10}},
+        rotation=90,
+        origin={-50,0})));
+  Modelica.StateGraph.StepWithSignal twoOn
+    "Status of two chillers on"
+    annotation (Placement(
+        transformation(
+        extent={{10,-10},{-10,10}},
+        rotation=90,
+        origin={-50,-70})));
+  Modelica.StateGraph.Transition offToOne(
```

Comment:
better change all this transition to the one with signal `Modelica.StateGraph.TransitionWithSignal`. And calculate the boolean condition through graphical implementation.